src/libraries/System.Runtime/tests/System.Runtime.Tests/System/ArrayTests.cs

```diff
@@ -1597,13 +1598,36 @@ public static void Copy_SourceAndDestinationNeverConvertible_ThrowsArrayTypeMism
             Assert.Throws<ArrayTypeMismatchException>(() => sourceArray.CopyTo(destinationArray, (long)destinationArray.GetLowerBound(0)));
         }
 
+        [Fact]
+        [SkipOnMono("Mono does not support pointer compatibility in Array.Copy")]
```

Comment:
Could you please open an issue on this and disable the test against this issue?

Reply:
Is it something we'd ever want to support on mono?

Reply:
We do not want to have behavior differences between different runtime. Every behavior difference between runtime is a bug. We may choose to not fix some of these bugs, but I do not see a good reason for it here.

For example, similar Mono-specific issue in casting logic was fixed just a few days ago: #103841

Reply:
Currently the behavior is different among all three runtimes. NativeAOT allows conversion between *any* pointers.

The current coreclr behavior is really complex to support. Can we make a breaking change instead to support only exactly same pointers?

Reply:
> The current coreclr behavior is really complex to support.

Why is it complex to support?

> Can we make a breaking change instead to support only exactly same pointers?

I do not see how we would justify this breaking change.

Reply:
Opened #104197 for mono.